src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/ArchiveAssemblyHelper.cs

```diff
@@ -187,6 +187,27 @@ public List<string> ListArchiveContents (string storeEntryPrefix = DefaultAssemb
 			return entries;
 		}
 
+		public int GetNumberOfAssemblies (bool countAbiAssembliesOnce = true, bool forceRefresh = false)
+		{
+			List<string> contents = ListArchiveContents (assembliesRootDir, forceRefresh);
+			var dlls = contents.Where (x => x.EndsWith (".dll", StringComparison.OrdinalIgnoreCase));
+
+			if (!countAbiAssembliesOnce) {
+				return dlls.Count ();
+			}
+
+			var cache = new HashSet<string> (StringComparer.OrdinalIgnoreCase);
+			return dlls.Where (x => {
```

Comment:
Why `dlls.Where()` + `HashSet<string>` instead of [`dlls.Select(n => Path.GetFileName(n)).Distinct(StringComparer.OrdinalIgnoreCase)`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct?view=net-5.0#System_Linq_Enumerable_Distinct__1_System_Collections_Generic_IEnumerable___0__System_Collections_Generic_IEqualityComparer___0__)?

Reply:
I don't like LINQ too much and the `HashSet<string>` route felt faster (one pass over the set instead of two)